### PR TITLE
Fix //src/main/java/com/google/devtools/build/lib:gen_command-line-reference and add tests.

### PR DESCRIPTION
**Pull request by jin**
09924273efc7050a2e342cd89b823a619ca073f8 adds a new required --failure_detail_out flag, which broke the docgen genrule because its invoking the Bazel server directly without the flag.

We should revisit this direct usage of the server, and look at using the Bazel launcher directly. https://github.com/bazelbuild/bazel/issues/11095#issuecomment-611063162

For now, manually add the flag to get our documentation pipeline working again.

Fixes https://github.com/bazelbuild/bazel/issues/11095

### CODE REVIEW

**Comment by jin** [comment]
cc @anakanemison (can't add you as a reviewer for some reason, maybe you're not in the bazelbuild GitHub org?

File: src/test/java/com/google/devtools/build/docgen/generated_docs_build_test.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# A smoke file test for generated doc files.
+
+set -euo pipefail
+
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
```
Comment (aiuto) [inline src/test/java/com/google/devtools/build/docgen/generated_docs_build_test.sh:9]: This is obviously unneeded, but it also makes sense to copy over a tricky preamble like this without changes. So I'm calling it out as OK, in spite of it looking funny.

File: src/test/java/com/google/devtools/build/docgen/generated_docs_build_test.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+for out in $EXPECTED_ZIPS; do
+    filepath="$(rlocation io_bazel/src/main/java/com/google/devtools/build/lib/$out)"
+    # Test that the file has a corrected ZIP structure.
+    zipinfo "$filepath" 1>/dev/null
+done
+
+COMMAND_LINE_REF_FILE="$(rlocation io_bazel/src/main/java/com/google/devtools/build/lib/command-line-reference.html)"
+# Test that the file contains (at least) the title string.
+grep "Command-Line Reference" "$COMMAND_LINE_REF_FILE" 1>/dev/null
```
Comment (aiuto) [inline src/test/java/com/google/devtools/build/docgen/generated_docs_build_test.sh:28]: grep -q  ....
will just set an exit status without doing any output.

Comment (jin) [reply src/test/java/com/google/devtools/build/docgen/generated_docs_build_test.sh]: Nice, done.

**Pull request closed without merging**